spec/timezone.html

```diff
@@ -536,32 +536,37 @@ <h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It parses the argument as a numeric UTC offset string and returns a signed integer representing that offset as a number of nanoseconds.</dd>
+      </dl>
       <emu-alg>
-        1. Assert: Type(_offsetString_) is String.
-        1. If _offsetString_ does not satisfy the syntax of a |TimeZoneNumericUTCOffset| (see <emu-xref href="#sec-temporal-iso8601grammar"></emu-xref>), then
-          1. Throw a *RangeError* exception.
-        1. Let _sign_, _hours_, _minutes_, _seconds_, and _fraction_ be the parts of _offsetString_ produced respectively by the |TimeZoneUTCOffsetSign|, |TimeZoneUTCOffsetHour|, |TimeZoneUTCOffsetMinute|, |TimeZoneUTCOffsetSecond|, and |TimeZoneUTCOffsetFraction| productions, or *undefined* if not present.
-        1. Assert: _sign_ is not *undefined*.
-        1. Assert: _hours_ is not *undefined*.
-        1. If _sign_ is the code unit 0x002D (HYPHEN-MINUS) or 0x2212 (MINUS SIGN), then
-          1. Set _sign_ to -1.
+        1. Let _parseResult_ be ParseText(StringToCodePoints(_offsetString_), |TimeZoneNumericUTCOffset|).
+        1. If _parseResult_ is a List of errors, throw a *RangeError* exception.
+        1. Let each of _sign_, _hours_, _minutes_, _seconds_, and _fSeconds_ be the source text matched by the respective |TimeZoneUTCOffsetSign|, |TimeZoneUTCOffsetHour|, |TimeZoneUTCOffsetMinute|, |TimeZoneUTCOffsetSecond|, and |TimeZoneUTCOffsetFraction| Parse Node contained within _parseResult_, or an empty sequence of code points if not present.
+        1. Assert: _sign_ is not empty.
+        1. If _sign_ contains the code point U+002D (HYPHEN-MINUS) or U+2212 (MINUS SIGN), then
+          1. Let _factor_ be -1.
         1. Else,
-          1. Set _sign_ to 1.
-        1. Set _hours_ to ! ToIntegerOrInfinity(_hours_).
-        1. Set _minutes_ to ! ToIntegerOrInfinity(_minutes_).
-        1. Set _seconds_ to ! ToIntegerOrInfinity(_seconds_).
-        1. If _fraction_ is not *undefined*, then
-          1. Set _fraction_ to the string-concatenation of the previous value of _fraction_ and the string *"000000000"*.
-          1. Let _nanoseconds_ be the String value equal to the substring of _fraction_ from 1 to 10.
-          1. Set _nanoseconds_ to ! ToIntegerOrInfinity(_nanoseconds_).
+          1. Let _factor_ be 1.
+        1. Assert: _hours_ is not empty.
+        1. Let _hoursMV_ be ! ToIntegerOrInfinity(CodePointsToString(_hours_)).
+        1. Let _minutesMV_ be ! ToIntegerOrInfinity(CodePointsToString(_minutes_)).
+        1. Let _secondsMV_ be ! ToIntegerOrInfinity(CodePointsToString(_seconds_)).
+        1. If _fSeconds_ is not empty, then
+          1. Let _fSecondsDigits_ be the substring of CodePointsToString(_fSeconds_) from 1.
+          1. Let _fSecondsDigitsExtended_ be the string-concatenation of _fSecondsDigits_ and *"000000000"*.
+          1. Let _nanosecondsDigits_ be the substring of _fSecondsDigitsExtended_ from 1 to 10.
```

Comment:
(actually, same here? we already chop the decimal point off in line 563?)

Reply:
```suggestion
          1. Let _nanosecondsDigits_ be the substring of _fSecondsDigitsExtended_ from 0 to 9.
```